### PR TITLE
tests: Bluetooth: tester: Add initial support for HAP

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This adds initial support for Hearing Aid Profile BTP service commands.

### CODE REVIEW

File: subsys/bluetooth/audio/has_client.c [hunk 1]
```diff
@@ -1000,3 +1000,24 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
...
+		return -ENODEV;
+	}
+
+	info->has_hearing_aid_features_handle = inst->features_subscription.value_handle;
+	info->has_control_point_handle = inst->control_point_subscription.value_handle;
+	info->has_active_preset_index_handle = inst->active_index_subscription.value_handle;
+
+	return 0;
+}
+#endif /* CONFIG_BT_TESTING */
```
Comment (Thalley) [inline subsys/bluetooth/audio/has_client.c:1023]: What we have done for the other implementations is to move the definition of e.g. `struct has_inst` to the internal header files, and simply included them in the tests. 

This avoids having test specific functions in the implementations. 

Did you consider this approach, and if so, why did you go with this approach?

File: subsys/bluetooth/audio/has_internal.h [hunk 2]
```diff
@@ -8,6 +8,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/bluetooth/bluetooth.h>
```
Comment (Thalley) [inline subsys/bluetooth/audio/has_internal.h:11]: Was this to have the `struct bt_conn`? If so, shouldn't this have been `conn.h`?

File: tests/bluetooth/tester/src/btp_hap.c [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+	tester_event(BTP_SERVICE_ID_HAP, BT_HAP_EV_HAUC_DISCOVERY_COMPLETE, &ev, sizeof(ev));
+}
+
+static void has_client_discover_cb(struct bt_conn *conn, int err, struct bt_has *has,
+				   enum bt_has_hearing_aid_type type, enum bt_has_capabilities caps)
+{
+	LOG_DBG("conn %p err %d", (void *)conn, err);
+
+	if (err != 0 && err != BT_ATT_ERR_ATTRIBUTE_NOT_FOUND) {
+		LOG_DBG("Discover remote ASEs failed: %d", err);
```
Comment (Thalley) [inline tests/bluetooth/tester/src/btp_hap.c:118]: This isn't discovering ASEs, right?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/has_client.c (hunk 1 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/has_client.c:1023]: No, I did not consider this, but it's a better approach actually. Will do that.

**Review by Thalley — APPROVED** [review]

**Review by larsgk — APPROVED** [review]

**Pull request merged**